src/IO/S3/Client.cpp

```diff
@@ -987,6 +987,11 @@ ClientFactory::ClientFactory()
     /// We don't need to initialize TLS, because we use PocoHTTPClientFactory
     aws_options.ioOptions.tlsConnectionOptions_create_fn = []() { return nullptr; };
 
+#if USE_JEMALLOC
+    memory_manager = std::make_unique<AwsNodumpMemoryManager>();
+    aws_options.memoryManagementOptions.memoryManager = memory_manager.get();
```

Comment:
This means we're using the NoDump allocator for all AWS-related things, right?

Reply:
For all internal allocations in the S3 client, yes